From: Kokan <kokaipeter@gmail.com>
Date: Fri, 24 Jul 2020 11:20:44 +0200
Subject: syslogformat: skip cisco sequence id parsing

If the parsing of cisco sequence id is failing, re-submit the string part it
tried and skip setting sequence id.

Signed-off-by: Kokan <kokaipeter@gmail.com>
Origin: https://github.com/syslog-ng/syslog-ng/commit/849305fc35f4097201183b25200f2af3ad56d404
---
 modules/syslogformat/syslog-format.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/modules/syslogformat/syslog-format.c
+++ b/modules/syslogformat/syslog-format.c
@@ -198,10 +198,12 @@
   while (left && *src != ':')
     {
       if (!isdigit(*src))
-          return FALSE;
-        _process_any_char(self, &src, &left);
+        return FALSE;
+      if (!_process_any_char(self, &src, &left))
+        return FALSE;
     }
-  _process_any_char(self, &src, &left);
+  if (!_process_any_char(self, &src, &left))
+    return FALSE;
 
   /* if the next char is not space, then we may try to read a date */
 
